02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,83 @@
+const { people } = require('../data');
+
+const addPerson = (req, res) => {
+    const { name } = req.body
+    if (!name) {
+        return res
+            .status(400)
+            .json({ success: false, message: 'Please provide a name' })
+    }
+    people.push({ id: people.length + 1, name: name })
+    res.status(201).json({ success: true, name: name })
+}
+
+const getPeople = (req, res) => {
+	res.json(people)
+}
+
+const getPerson = (req, res) => {
+    const id = parseInt(req.params.id)
+    if(!checkForValue(res, id)) {
+        return;
+    }
+    const personById = findPersonById(res, id);
+    if(!personById) {
+        return;
+    }
+       
+    res.status(200).json({ success: true, message: personById })
+}
+
+const updatePerson = (req, res) => {
+    const id = parseInt(req.params.id);
+    const updatingName = req.body.updatingName;
+    if(!checkForValue(res, id)) {
+        return;
+    }
+    const personById = findPersonById(res, id);
+    if(!personById) {
+        return;
+    }
+    console.log(updatingName)
+    console.log(people[id - 1])
+    people[id - 1].name = updatingName;
+    res.json({ success: true, message: people[id - 1] })
+}
+
+const deletePerson = (req, res) => {
+    const id = parseInt(req.params.id);
+    if(!checkForValue(res, id)) {
+        return;
+    }
+    const newPeople = people.filter(personObj => {
+        return personObj.id !== id
+    })
+    res.json({ message: "success", newPeople })
+}
+
+function checkForValue(res, id) {
```

Comment:
## Optional Improvement
[Understandability]
_Tldr; This function is actually more confusing than helpful. Removing it would actually help the code be more understandable._

Implemented as it stands this also sort of has a side effect (see my other comment on side effects). Overall this one is a little different because the function itself is really only responsible for handling the input sanitization so its not as much of a side effect because setting the 400 is very tightly coupled to the task of sanitization.

The more salient point here isn't about function cleanliness or side-effects, it's really about the value of this function.
Without this function, your code looks like this:
```js
const updatePerson = (req, res) => {
    const id = parseInt(req.params.id);
    const updatingName = req.body.updatingName;
    if(!id) {
        res.status(400).json({ success: false, message: 'please provide an id number' });
        return;
    }
    ...
}
```

This is honestly much more clear when reading it than your current implementation.

Generally speaking, you want to add an abstraction (like a function) if the code is complex enough that it can't easily be understood or used. The abstraction helps you break it down so its less overwhelming and using good names keeps it readable and clear. It's worth noting this isn't about the size of the function, it's about the complexity. I have advised many times for developers to make functions from 1 line of code, cause by making a function they could give that function a clear name that made 1 line of super complex code incredibly easy to understand and use.
The other reason (which is more contentious) is to avoid lots of duplicate code which can cause maintenance problems. 
High level though, the abstraction itself actually adds complexity so it needs to offset some complexity to be worth it (also if its poorly named it can add a whole lot of complexity).

In my opinion here the complexity without the function is incredibly small, I can see at a glance exactly how it works and why its that way. Without the function I actually have a lot of confidence in how the code works and how I can change it that is lost once the function is introduced. The function does save one instance of repeat code, but its not critical enough to actually save anything notable in terms of maintenance and DRY (Don't Repeat Yourself) should be treated as a loose guideline rather than a hard rule - there's a ton of evidence and programmers who believe that DRY is more harmful than good.